bin/qq

```diff
@@ -3,23 +3,36 @@
 require_relative "../lib/quiet_quality"
 
 tool_names = ARGV.empty? ? QuietQuality::Tools::AVAILABLE.keys : ARGV.map(&:to_sym)
+outcomes = []
 
 tool_names.each do |tool_name|
   tool = QuietQuality::Tools::AVAILABLE.fetch(tool_name)
   warn "================ Running #{tool_name}"
   runner = tool::Runner.new
   outcome = runner.invoke!
+  outcomes << outcome
 
   if outcome.success?
     warn "No problems found!\n\n"
   else
     parser = tool::Parser.new(outcome.output)
     messages = parser.messages
+
     warn "#{messages.count} messages\n"
     messages.each do |msg|
       warn "  #{msg.path}:#{msg.start_line}-#{msg.stop_line}    #{msg.rule}"
       warn "      #{msg.body}"
     end
     warn "\n"
+
+    if ENV.fetch("ANNOTATOR", nil) == "github_stdout"
+      QuietQuality::Annotators::GithubStdout.new.annotate!(messages)
+    end
+
   end
 end
+
+if outcomes.any? { |o| o.failure? }
+  warn "failures detected in one or more tools"
```

Comment:
Maybe in the future we can be _smarter_ about presenting something like this. Possibly we can say which tool failed?

Reply:
Sorry, I missed this comment somehow (I blame the github app, I like the web interface _way_ better). Currently, that information is presented earlier, since we iterate the tools and produce output for each one - it looks like this:

```
❯ bin/qq
================ Running rspec
1 messages
  spec/quiet_quality/message_spec.rb:24-24    RSpec::Expectations::ExpectationNotMetError
      expected #<QuietQuality::Message:0x00007fd6fe71c848 @attrs={"path"=>"/foo/path.rb", "body"=>"fake message", "s...e", @start_line=10, @stop_line=14, @annotated_line=12, @level="serious", @rule="You Shall Not Pass"> not to be a kind of QuietQuality::Message

================ Running rubocop
No problems found!

================ Running standardrb
2 messages
  spec/quiet_quality/message_spec.rb:33-34    Layout/EmptyLinesAroundBlockBody
      Extra empty line detected at block body beginning.
  spec/quiet_quality/message_spec.rb:34-35    Layout/EmptyLines
      Extra blank line detected.

failures detected in one or more tools
```

That last line is really just because, after you run a tool, you typically read the last line to see if it worked. But the overall presentation isn't really set yet - this script as it exists was just a placeholder so I could execute the code easily while building it. #36, #37, and #39 are all concerned with shaping the actual output system, and the `--light` flag described in #37 actually describes pretty much what you're talking about. I'll add another ticket though, for shaping the "normal" output more intentionally - I've been waiting that on the existence of the Logger from #39